spec-tests/src/test/java/org/candlepin/spec/ExportSpecTest.java

```diff
@@ -0,0 +1,649 @@
+
+import org.apache.commons.codec.binary.Base64;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+import org.junit.jupiter.api.TestInstance.Lifecycle;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.time.OffsetDateTime;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.stream.Collectors;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipFile;
+
+@SpecTest
+class ExportSpecTest {
+    private static final String EXPORT_JOB_KEY = "ExportJob";
+    private static final String RECORD_CLEANER_JOB_KEY = "ImportRecordCleanerJob";
+    private static final String UNDO_IMPORTS_JOB_KEY = "UndoImportsJob";
+    /** The following are directory location paths within a manifest */
+    private static final String EXPORT_PATH = "export/";
+    private static final String PRODUCTS_PATH = "export/products/";
+    private static final String ENTITILEMENTS_PATH = "export/entitlements/";
+    private static final String ENTITILEMENT_CERTIFICATES_PATH = "export/entitlement_certificates/";
+    private static final String DISTRIBUTOR_VERSION_PATH = "export/distributor_version/";
+    private static final String CONSUMER_TYPE_PATH = "export/consumer_types/";
+    private static final String CDN_PATH = "export/content_delivery_network/";
+    private static final String RULES_PATH = "export/rules2/";
+
+    @Test
+    @DisplayName("should allow manifest creation with read only user")
+    void shouldAllowManifestCreationWithReadOnlyUser() throws Exception {
+        ApiClient client = ApiClients.admin();
+        OwnerDTO owner = client.owners().createOwner(Owners.random());
+        ConsumerDTO readOnlyConsumer = createReadOnlyConsumer(client, owner);
+
+        File readOnlyManifest = createExport(client, readOnlyConsumer.getUuid(), null);
+        ZipFile readOnlyExport = ExportUtil.getExportArchive(readOnlyManifest);
+        String path = EXPORT_PATH + "consumer.json";
+
+        ConsumerDTO actualConsumer = ExportUtil
+            .deserializeJsonFile(readOnlyExport, path, ConsumerDTO.class);
+        assertEquals(readOnlyConsumer.getUuid(), actualConsumer.getUuid());
+        assertEquals(readOnlyConsumer.getName(), actualConsumer.getName());
+
+        assertEquals(owner.getId(), actualConsumer.getOwner().getId());
+        assertEquals(owner.getKey(), actualConsumer.getOwner().getKey());
+        assertEquals(owner.getDisplayName(), actualConsumer.getOwner().getDisplayName());
+    }
+
+    @Nested
+    @DisplayName("Standard Export Tests")
+    @TestInstance(Lifecycle.PER_CLASS)
```

Comment:
Why the test needs the PER_CLASS lifecycle? It can break parallelization by making tests share instance properties.

Reply:
Good point! The PER_CLASS lifecycle should not be necessary and we can remove it.